docs/en/advanced_guides/detection_transformer.md

```diff
@@ -0,0 +1,225 @@
+
+The aforementioned operations support the transformation of a single-level feature. For transformation of multi-level features, more information should be recorded.
+
+The multi-level features is actually a tuple of multiple image features. The format of the `l`-th level image feature is `(B, C, H_l, W_l)`, where `H_l` and `W_l` are height and width, respectively. The tuple of image features is also transformed into a sequence feature. Specifically, each image feature is transformed into a sequence feature of length `N_l = H_l x W_l` by flattening and permuting. Then, the obtained sequence features are concatenate on the dimension `N = N_1 + ... + N_L`.
+
+```python
+feat_flatten = []
+for lvl, feat in enumerate(mlvl_feats):
+    batch_size, c, h, w = feat.shape
+    # [bs, c, h_lvl, w_lvl] -> [bs, h_lvl*w_lvl, c]
+    feat = feat.view(batch_size, c, -1).permute(0, 2, 1)
+    feat_flatten.append(feat)
+feat_flatten = torch.cat(feat_flatten, 1)
+```
+
+When processing multi-level features, the level embeddings are usually added to the positional embeddings to distinguish the feature levels. The sum is represented with `lvl_pos_embed`.
+
+To support more operations on multi-scale features, some extra information should be introduced. For example, the feature `spatial shape` on each level, the `lvl_start_index` (the start sequence indexes of each feature level), and so on. The `spatial shape` and `lvl_start_index` can be used to restore the sequence feature of `(B, N, C)` to the tuple of multi-scale features of `B, C, H_l, W_l`. They can also support special multi-scale feature interaction operations, such as Deformable Attention.
+
+***( Add a figure here, to illustrate multi-scale features \<--> sequence feature, what is spatial_shape and lvl_start_index)***
+
+#### Positional embedding of DETRs
+
+For the sequence data, both the position of each element in the sequence and the arrangement of the elements are essential for the semantic characterization of the data; For the image data, both the position of each pixel on the whole image and the spatial arrangement of the pixels are also critical for understanding image semantic information.
+
+The attention calculation procedure of Transformer is usually as follow: A query embedding calculates self attentions with the key embeddings. Then, the value embeddings are weighted with the attentions and aggregated into the output embedding. In the attention calculations, each element in the sequence is independent, and their position information is lost. Besides, the attention calculations are permutation-invariant. Therefore, the position encodings are usually embedded for queries, keys, and values before calculating attentions as positional embeddings.
```

Comment:
`the position encodings are usually embedded for queries, keys, and values`
There is no position embedding for values